packages/discord.js/src/structures/ButtonComponent.js

```diff
@@ -1,11 +1,64 @@
 'use strict';
 
-const { ButtonComponent: BuildersButtonComponent } = require('@discordjs/builders');
-const Transformers = require('../util/Transformers');
+const Component = require('./Component');
 
-class ButtonComponent extends BuildersButtonComponent {
-  constructor(data) {
-    super(Transformers.toSnakeCase(data));
+/**
+ * Represents a button component
+ * @extends {Component}
+ */
+class ButtonComponent extends Component {
+  /**
+   * The style of this button
+   * @type {ButtonStyle}
+   * @readonly
+   */
+  get style() {
+    return this.data.style;
+  }
+
+  /**
+   * The label of this button
+   * @type {?string}
+   * @readonly
+   */
+  get label() {
+    return this.data.label;
+  }
+
+  /**
+   * The emoji used in this button
+   * @type {?APIMessageComponentEmoji}
+   * @readonly
+   */
+  get emoji() {
+    return this.data.emoji;
+  }
+
+  /**
+   * Whether this button is disabled
+   * @type {?boolean}
+   * @readonly
+   */
+  get disabled() {
+    return this.data.disabled;
+  }
+
+  /**
+   * The custom id of this button (only defined on non-link buttons)
+   * @type {?string}
+   * @readonly
+   */
+  get customId() {
+    return this.data.custom_id;
+  }
+
+  /**
+   * The URL of this button (only defined on link buttons)
+   * @type {?string}
+   * @readonly
+   */
+  get url() {
+    return this.data.url;
```

Comment:
```suggestion
  /**
   * The custom id of this button
   * @type {string}
   * @readonly
   */
  get customId() {
    return this.data.custom_id;
```

You can't receive interaction from a link button so url doesn't make sense here.

Reply:
These aren't solely for interactions though? ButtonComponents can be LINK type and exist in Messages. Which your bot absolutely sees/receives/handles.

Reply:
Yeah. I forgot sorry